pkg/block/block.go

```diff
@@ -120,6 +131,65 @@ func Download(ctx context.Context, bucket objstore.Bucket, id ulid.ULID, dst str
 	return nil
 }
 
+// Upload uploads block from given block dir that ends with block id.
+// It makes sure cleanup is done on error to avoid partial block uploads.
+// It also verifies basic features of Thanos block.
+// TODO(bplotka): Ensure bucket operations have reasonable backoff retries.
+func Upload(ctx context.Context, bkt objstore.Bucket, bdir string) error {
```

Comment:
Do we need some extra logic to cleanup partial uploads of the context gets canceled mid-way?